crates/parcel/src/cache/mod.rs

```diff
@@ -0,0 +1,119 @@
+use std::path::Path;
+
+use heed::types::{Bytes, Str};
+use heed::{EnvFlags, EnvOpenOptions, RoTxn};
+
+use parcel_core::cache::Cache;
+
+type Database = heed::Database<Str, Bytes>;
+
+/// Implements a `lmdb` cache back-end using [`heed`].
+///
+/// This should change as we progress because we would want to:
+///
+/// * We want the cache to be able to store data and be opinionated about the serialization format
+///   That is, the current API of receiving and returning bytes is likely not what we want. Instead
+///   we want the cache to handle serializing entries.
+/// * Cache keys don't need to be strings; ideally they'll be strongly typed enums which we will
+///   implement efficient serialization into nice keys we can iterate and lookup efficiently
+/// * Entries should use binary serialization.
+/// * We don't need to allocate when returning the entries
+pub struct LMDBCache {
+  environment: heed::Env,
+  database: Database,
+}
+
+impl LMDBCache {
+  #[allow(unused)]
+  pub fn new(options: LMDBCacheOptions) -> anyhow::Result<Self> {
+    let rust_cache_path = Path::new(".parcel-cache/rust-cache");
```

Comment:
Should use `join`

Reply:
I think this works on windows - https://doc.rust-lang.org/std/path/struct.Path.html

Reply:
Interesting, I've had a lot of issues with it before but maybe it's from mixing the two